src/modules/auth/helper/fabricWallet.ts

```diff
@@ -146,4 +175,59 @@ export class FabricWallet {
       privateKey: enrollment.key.toBytes().replace(/\r/g, ''),
     };
   }
+
+  /**
+   * Retrieves the identity of a user from the wallet.
+   * @param token - The unique identifier for the user's identity.
+   * @returns The user's identity object, containing their credentials and MSP ID.
+   * @throws Error - If the provided token does not correspond to a valid user ID.
+   */
+  async getIdentity(token: string) {
+    const identity = await FabricWallet.wallet.get(token);
+    if (!identity) {
+      throw new Error(`Invalid User ID: ${token}`);
+    }
+    return identity;
+  }
+
+  // will be removed after testing the contract flow!!!
+  async callContract(callConractInputDto: CallContractInputDto) {
+    const { token, channelName, contractName, functionName, attrs } =
+      callConractInputDto;
+    const identity = await FabricWallet.wallet.get(token);
+    if (!identity) {
+      throw new Error(`Invalid admin ID: ${token}`);
+    }
+    const gateway = new Gateway();
+    const gatewayOpts: GatewayOptions = {
+      identity,
+      wallet: FabricWallet.wallet,
+      discovery: { enabled: true, asLocalhost: true },
+    };
+
+    try {
+      await gateway.connect(ccp, gatewayOpts);
+      const network = await gateway.getNetwork(channelName);
+      const contract = network.getContract(contractName);
+
+      // Invoke the contract function
+      const evaluateTx = await contract.evaluateTransaction(
+        functionName,
+        ...attrs,
+      ); // to GET something from the contract
+      console.log(
+        'Transaction has been evaluated, result: ',
+        evaluateTx.toString(),
+      );
+      const result = await contract.submitTransaction(functionName, ...attrs); // to POST something to the contract
+      console.log(
+        'Transaction has been submitted, result: ',
+        result.toString(),
+      );
+      return result.toString();
```

Comment:
One thing I forgot to do here was closing the opened gateway before returning from this function. I'm not sure if that is handled by default, but I think we should add `gateway.disconnect();` here.

Reply:
Noted